src/core/NavBar/NavBarLink.tsx

```diff
@@ -0,0 +1,60 @@
+import { NavLink } from 'react-router-dom';
+
+import NavBarLinkFlyout from './NavBarLinkFlyout';
+
+interface NavBarLinkProps {
+  to: string;
+  text: string;
+  icon: React.ReactNode;
+  expanded: boolean;
+  onActive: () => void;
+}
+
+export default function NavBarLink({
+  to,
+  icon,
+  text,
+  expanded,
+  onActive
+}: NavBarLinkProps) {
+  return (
+    <NavLink
+      to={to}
+      className={`relative h-12 shadow-sm group ${
+        expanded ? 'w-full' : 'w-12'
+      }`}>
+      {({ isActive }) => {
+        if (isActive) {
+          setTimeout(() => {
+            onActive();
+          }, 0);
+        }
+
+        return (
+          <>
+            <div
+              className={`flex-center w-full h-full 0 transition-all overflow-hidden hover:bg-gray-800 ${
+                isActive ? 'bg-gray-800' : 'bg-gray-70'
```

Comment:
We could move the classes (especially dynamic one) to some variable, and then use it here. It would make the code much cleaner.

Reply:
Well, I agree with you that long class names are somehow hard to read but the general idea od Tailwind is to have everything in one place and keep styles inline to make it faster to develop and easier to change (as you have styles directly assigned to specific HTML tags). I can extract the static part of the outermost `div` to a variable because dynamic parts are quire short and adding multiple variables for classes might have the opposite effect and a negative impact on the readability.